ignite/contrib/handlers/base_logger.py

```diff
@@ -147,26 +147,34 @@ class BaseLogger(metaclass=ABCMeta):
     """
 
     def attach(
-        self, engine: Engine, log_handler: Callable, event_name: Union[str, Events, CallableEventWithFilter]
+        self, engine: Engine, log_handler: Callable, event_name: Union[str, Events, CallableEventWithFilter, EventsList]
     ) -> RemovableEventHandle:
         """Attach the logger to the engine and execute `log_handler` function at `event_name` events.
 
         Args:
             engine (Engine): engine object.
             log_handler (callable): a logging handler to execute
             event_name: event to attach the logging handler to. Valid events are from
-                :class:`~ignite.engine.events.Events` or any `event_name` added by
-                :meth:`~ignite.engine.engine.Engine.register_events`.
+                :class:`~ignite.engine.events.Events` or class:`~ignite.engine.events.EventsList` or any `event_name`
+                added by :meth:`~ignite.engine.engine.Engine.register_events`.
 
         Returns:
             :class:`~ignite.engine.RemovableEventHandle`, which can be used to remove the handler.
         """
-        name = event_name
+        if isinstance(event_name, EventsList):
```

Comment:
I think it would be better to use `return engine.add_event_handler(event_name, log_handler, self, event_name)` for any type of events. Below for loop if considered without check `if name not in State.event_to_attr` is exactly what is done by `Engine.add_event_handler`.
I think we can simply generalize event checking like
```python
event_to_check = event_name if isinstance(event_name, EventsList) else [event_name, ]
for name in event_to_check:
    if name not in State.event_to_attr:
        raise RuntimeError(f"Unknown event name '{name}'")

return engine.add_event_handler(event_name, log_handler, self, event_name)
```

Reply:
That does not work.

The last argument of the method `add_event_handler` is a list of events. Internally in engine, there is a loop where the handler is attached at every event in the list. But the list is forwarded to each handler while just each event should be.

If there was no argument to the method, it would work. 

Thoughts ?

Reply:
Yes, you are right. In this case, my suggestion is not valid. OK, let's do it as you proposed.